book/07-git-tools/sections/rewriting-history.asc

```diff
@@ -8,6 +8,13 @@ This can involve changing the order of the commits, changing messages or modifyi
 
 In this section, you’ll cover how to accomplish these very useful tasks so that you can make your commit history look the way you want before you share it with others.
 
+[NOTE]
+====
+The cardinal rule with Git is that, since so much work is local within your clone, you have a great deal of freedom to rewrite your history _locally_.
+However, once you push your work, it is a different story entirely, and you should consider pushed work as final.
+In short, you should never push your work until you're happy with it and ready to share it with the rest of the world.
+====
```

Comment:
I agree with your goal here, which is to call out more prominently that history-rewriting is risky. I hate to be so prescriptive, though. This is absolutely good advice _in most circumstances,_ but there are times when rewriting an already-pushed commit is the right way forward.

I kind of want to pair a note like this, which cautions about rewriting shared commits, with a new section on what to do when shared history has been rewritten in `origin`, and now your team all have repos in weird states. The note should come at the top, and probably the new section at the end.

Do you think you're up to that on this branch? Or should we back this note out and make a separate issue out of it?

Reply:
I'll leave this note in here for now, but reword it to make it less draconian; then we can ponder how to expand on this more completely.